ml_tools/feature_generator.py

```diff
@@ -0,0 +1,161 @@
+import pandas as pd
+from abc import ABC, abstractmethod
+
+
+class FeatureGenerator(ABC):
+
+    # def __init__(self, data):
+    #     self._data = data.copy()
+
+    @abstractmethod
+    def generate_feature(self, data, column=None, **kwargs):
+        raise NotImplementedError("generate_feature must be implemented")
+
+
+# class GenericGenerator(FeatureGenerator):
```

Comment:
- [ ] Make sure to get rid of all your commented code.
- [ ] It looks like it, but everything is linted and PEP8 yes?